src/get-external-builtins.js

```diff
@@ -0,0 +1,96 @@
+/********************************************************************************
+ * Copyright (C) 2022 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+// @ts-check
+
+/**
+ * Clone and check-out the vscode external builtin extensions repositories, each to the version
+ * that's expected for the current vscode baseline (as defined in vscode's product.json)
+ */
+const { run, externalBuiltinsRepos, vscode } = require('./paths.js');
+const fs = require('fs-extra');
+
+CloneCheckoutExternalBuiltins();
+
+async function CloneCheckoutExternalBuiltins() {
+    /** Path of `vscode` product.json file */
+    const prod = vscode('product.json');
+    const content = fs.readFileSync(prod, 'utf-8');
+    /** 
+     * vscode product.json section where we find info about external builtins 
+     * @type ProductBuiltInExtensionEntry[]
+    */
+    const prodJsonExts = JSON.parse(content).builtInExtensions;
+
+    for (const entry of prodJsonExts) {
+        // extract repo directory name from last part of git repo URL and
+        const repoName = entry.repo.split("/").slice(-1)[0];
+        /** @type string[] */
+        const repoDirectories = fs.readdirSync(externalBuiltinsRepos());
```

Comment:
if we cleanup the directory `externalBuiltinsRepos` before entering the loop, we don't need to check if a plugin folder already exists, and if does, it means that the production.json has an errors and it's have duplicate entries, which is unlikely and not worth covering for it. 
any thoughts ?

Reply:
True. For now I have not implemented that suggested change, in favor of documenting how to start the process with a clean slate. I think this gives more flexibility, if we need to re-run the script, e.g. to re-clone one repo or checkout version corresponding to another `vscode` baseline.